stingray/fourier.py

```diff
@@ -1,15 +1,19 @@
 import copy
 import warnings
 from collections.abc import Iterable
+from typing import Optional
 
 import numpy as np
 from astropy.table import Table
+from astropy.timeseries.periodograms.lombscargle.implementations.utils import trig_sum
 
+from .events import EventList
```

Comment:
`fourier.py` should only contain funcitonality that is agnostic of Stingray classes like EventList, Lightcurve, etc.
Please only create here functions that work on simple numpy arrays (e.g. `events.time` or `lightcurve.counts`)

Reply:
Changed it to take numpy arrays and made the corresponding changes in commit  https://github.com/pupperemeritus/stingray/commit/8fdf44ae7fbe395050a384fd776c9c125ac6e996